src/connector/src/parser/additional_columns.rs

```diff
@@ -87,75 +88,172 @@ fn kafka_compatible_column_vec() -> Vec<(&'static str, CompatibleAdditionalColum
                     column_desc: ColumnDesc::named_with_additional_column(
                         name,
                         id,
-                        DataType::Int64,
+                        DataType::Varchar,
                         AdditionalColumnType::Offset,
                     ),
                     is_hidden: false,
                 }
             }),
         ),
-        // Todo(tabVersion): add header column desc
-        // (
-        //     "header",
-        //     Box::new(|id: ColumnId, name: &str| -> ColumnCatalog {
-        //         ColumnCatalog {
-        //             column_desc: ColumnDesc::named(name, id, DataType::List(
-        //
-        //             )),
-        //             is_hidden: false,
-        //         }
-        //     }),
-        // ),
+        (
+            "header", // type: struct<key varchar, value bytea>[]
+            Box::new(|id: ColumnId, name: &str| -> ColumnCatalog {
+                ColumnCatalog {
+                    column_desc: ColumnDesc::named_with_additional_column(
+                        name,
+                        id,
+                        DataType::List(get_kafka_header_item_datatype().into()),
+                        AdditionalColumnType::Header,
+                    ),
+                    is_hidden: false,
+                }
+            }),
+        ),
     ]
 }
 
+pub fn get_kafka_header_item_datatype() -> DataType {
+    let struct_inner = vec![("key", DataType::Varchar), ("value", DataType::Bytea)];
+    DataType::Struct(StructType::new(struct_inner))
+}
+
 fn pulsar_compatible_column_vec() -> Vec<(&'static str, CompatibleAdditionalColumnsFn)> {
-    vec![(
-        "key",
-        Box::new(|id: ColumnId, name: &str| -> ColumnCatalog {
-            ColumnCatalog {
-                column_desc: ColumnDesc::named_with_additional_column(
-                    name,
-                    id,
-                    DataType::Bytea,
-                    AdditionalColumnType::Key,
-                ),
-                is_hidden: false,
-            }
-        }),
-    )]
+    vec![
+        (
+            "key",
```

Comment:
If the upstream schema also has a column called `key` / `partition` / `offset` will it conflict?

Reply:
no, these parts are configured in the `include` clause.